src/scripts/Build-Workbook.ps1

```diff
@@ -24,14 +24,56 @@ $Debug = $DebugPreference -eq "Continue"
 $outDir = Join-Path .. .. release "$Workbook-workbook"
 $workbookDir = Join-Path .. workbooks $Workbook
 
+# Define workbooks metadata
+$workbooksMetdata = @(
+  [PSCustomObject]@{
+    optimization = @{
+      "Product"     = "Azure Advisor"
+      "GalleryName" = "Cost Optimization"
+    }
+  }
+)
+
 if (-not (Test-Path $workbookDir)) {
-    return
+  return
+}
+
+# Get all directories within the workbook folder
+$templates = Get-ChildItem -Path $workbookDir -Directory
+
+# Check if any templates were found
+if ($templates) {
+  $workbookTemplate = Join-Path $workbookDir "workbook.json"
+  $newTemplate = "$outDir/workbook.json"
+  $workbookProduct = $workbooksMetdata.$workbook["Product"]
+  $workbookGalleryName = $workbooksMetdata.$workbook["GalleryName"]
+  ## Create a new template
+  Copy-Item $workbookTemplate $newTemplate -Force
+  $newWorkbookContent = Get-Content $newTemplate | ConvertFrom-Json
+
+  ## Inject contents of each sub-template
+  foreach ($template in $templates) {
+    $templateName = $template.Name
+    $tempTemplate = Get-Content "$workbookDir/$templateName/$templateName.workbook" -Raw
+    $templateJson = $tempTemplate | ConvertFrom-Json
+    $templateObjects = ($templateJson.items.content).items
+    $templateLoadString = "community-Workbooks/$workbookProduct/$workbookGalleryName/$templateName"
+    $templateContent = $newWorkbookContent.items.content.items | Where-Object { $_.content.groupType -eq 'template' } | Select-Object -ExpandProperty content | Where-Object { $_.loadFromTemplateId -eq $templateLoadString }
+    $templateContent.loadFromTemplateId = '""'
+    $templateObjects | ForEach-Object {
+      $templateContent.items += $_
+    }
+  }
```

Comment:
Can't we find the nested templates and then pull the content rather than looping thru the folders? Then we shouldn't need the hard-coded folders, right?
```suggestion
    $templateContent = $newWorkbookContent.items.content.items `
    | Where-Object { $_.content.groupType -eq 'template' } `
    | Select-Object -ExpandProperty content `
    | Where-Object { $_.loadFromTemplateId.StartsWith("community-Workbooks") } `
    | ForEach-Object {
        $templateName = $_.loadFromTemplateId.Split('/')[3]
        $tempTemplate = Get-Content "$workbookDir/$templateName/$templateName.workbook" -Raw
        $templateJson = $tempTemplate | ConvertFrom-Json
        $templateObjects = ($templateJson.items.content).items
        $templateObjects | ForEach-Object {
            $templateContent.items += $_
        }
    }
```